subprojects/shipkit/src/main/groovy/org/shipkit/internal/notes/contributors/IgnoredContributor.java

```diff
@@ -0,0 +1,35 @@
+package org.shipkit.internal.notes.contributors;
+
+import org.shipkit.internal.notes.model.Contributor;
+import org.shipkit.internal.notes.util.Predicate;
+
+import java.util.Collection;
+import java.util.Collections;
+
+class IgnoredContributor implements Predicate<Contributor> {
+
+    private final Collection<String> ignoredContributors;
+
+    static IgnoredContributor none() {
+        return new IgnoredContributor(Collections.<String>emptyList());
+    }
+
+    static IgnoredContributor of(Collection<String> ignoredContributors) {
+        return new IgnoredContributor(ignoredContributors);
+    }
+
+    private IgnoredContributor(Collection<String> ignoredContributors) {
+        this.ignoredContributors = ignoredContributors;
+    }
+
+    @Override
+    public boolean isTrue(Contributor contributor) {
```

Comment:
Let's make this class a little more generic by accepting a string here, and explaining what format exactly this contributor string should be. This way we can reuse this class in IgnoredCommit instead of duplicating the for loop.

Reply:
It's better to reuse the code, but I'm not sure if `IgnoredContributor` should be parametrized with String. We would move some logic for ignoring contributor outside the class, because it's caller, who  would have to know which part of contributor should be checked. It also closes the class from extension of ignoring rules.

I would propose extending class `IgnoredContributor` with method `#isTrue(String contributorLogin)` and provide `IgnoredContributor` explicitly (by exact class) to `IgnoredCommit`

Reply:
I'm not sure if I follow. 
Let me sum it up:
- method IgnoredContributor.isTrue would be overloaded, another instance of it would take String as argument instead of Contributor object
- IgnoredCommit uses IgnoredContributor instead of duplicating logic

Let me know if I got it right :)

> We would move some logic for ignoring contributor outside the class, because it's caller, who would have to know which part of contributor should be checked.

You are right, using an object like Contributor generally makes it easier to use it - we have to only document Contributor class (what each field means exactly), and stick to this convention when we create this object and when we use it. In this case though we would have to create DefaultContributor in IgnoredCommit, which can be done but it makes it more complicated and harder to maintain (if Contributor grows, you have to remember about creating it with all necessary fields). This idea with overloading method sounds good! We can support both String (with a clear explanation what format it is in) and Contributor.

> It also closes the class from extension of ignoring rules.

I don't think it does that. It's not public API, it's easy to extend a method with additional argument in a scope of the single project. If we have more of them, we can stick to using Contributor itself and pass the problem of creating this object to the caller.

Reply:
Yes, you got it right!

>  If we have more of them, we can stick to using Contributor itself and pass the problem of creating this object to the caller.

I agree. I will overload method for now and if it would need extension in the future then someone will have to think about using `Contributor` only.

Thank you!